File: pkg/models/address.go
package models

// AddressModel -
type AddressModel struct {
	Street  string `bson:"street" json:"street"`
	City    string `bson:"city" json:"city"`
	ZipCode string `bson:"zipCode" json:"zipCode"`
	State   string `bson:"state" json:"state"`
	Country string `bson:"country" json:"country"` // CountryCode
}

// NewAddressModel -
func NewAddressModel() *AddressModel {
	return &AddressModel{}
}

File: web/src/components/generic/address.js
import React from "react"
import { useTranslation } from "react-i18next"

export const Address = ({ address }) => {
  const { t } = useTranslation()

  return (
    <div className="address">
      {address?.street && <div>{address?.street}</div>}
      {address?.city && <div>{address?.city}</div>}
      {address?.zipCode && <div>{address?.zipCode}</div>}
      {address?.state && <div>{address?.state}</div>}
      {address?.country && <div>{t(`countries:${address?.country}`)}</div>}
    </div>
  )
}
